webapp/src/components/SettingsPage/SettingsPage.js

```diff
@@ -46,72 +38,54 @@ export default class SettingsPage extends React.PureComponent {
     }
   }
 
-  handleMortgageForManaApproval = (event, data) => {
-    const manaToApprove = data.checked ? getManaToApprove() : 0
-    this.props.onAuthorizeMortgageForMana(manaToApprove)
+  handleTokenAllowance = (checked, contractName, tokenContractName) => {
+    const amount = checked ? getTokenAmountToApprove() : 0
+    this.props.onAllowToken(amount, contractName, tokenContractName)
   }
 
-  handleMortgageForRCNApproval = (event, data) => {
-    const rcnToApprove = data.checked ? getRCNToApprove() : 0
-    this.props.onAuthorizeMortgageForRCN(rcnToApprove)
+  handleTokenApproval = (checked, contractName, tokenContractName) => {
+    this.props.onApproveToken(checked, contractName, tokenContractName)
   }
 
-  getApproveTransaction() {
-    // Transactions are ordered, the last one corresponds to the last sent
-    const { approveManaTransactions } = this.props.wallet
-    return approveManaTransactions[approveManaTransactions.length - 1]
-  }
+  getPendingTransactions() {
```

Comment:
Should I leave this here or on the container?

Reply:
if this component only uses the pending transactions let's filter them in the container (we could use a selector for it) and pass only those down to this component